### ISSUE TITLE
pnpm outdated option for showing outdated transitive dependencies

### ISSUE THREAD

**Issue by segevfiner** [pnpm/pnpm#7323]
### Contribution

- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

I would like to be able to see all outdated dependencies, including transitive dependencies.

### Describe the solution you'd like

For `pnpm outdated` to support a flag to show outdated transitive dependencies.

### Describe the drawbacks of your solution

More code in pnpm?

### Describe alternatives you've considered

Try to `pnpm update` and check what changed, not very convenient...

**Comment by kjkent** [comment]
Agreed! Would be great to see this, I was searching for an answer and ended up making a discussion comment here: https://github.com/orgs/pnpm/discussions/4217#discussioncomment-9992374

From that comment:
>As far as I can tell, the only way to find which direct dependencies are responsible is to run pnpm up, then pnpm ls --depth <unknown number>, grep the tree for each of the outdated deps, then follow the tree back up to the direct dependency that pulled them in.
